auditbeat/module/auditd/audit_linux.go

```diff
@@ -175,6 +177,13 @@ func (ms *MetricSet) Run(reporter mb.PushReporterV2) {
 						ms.updateKernelLostMetric(status.Lost)
 					} else {
 						ms.log.Error("get status request failed:", err)
+						if err = client.Close(); err != nil {
+							ms.log.Errorw("Error closing audit monitoring client", "error", err)
+						}
+						client, err = libaudit.NewAuditClient(nil)
+						if err != nil {
+							ms.log.Errorw("Failure creating audit monitoring client", "error", err)
```

Comment:
If this one fails then I think `client == nil` which will cause problems on subsequent iterations.

Reply:
D'oh! Good catch